api/src/jmh/java/io/opentelemetry/trace/propagation/HttpTraceContextInjectBenchmark.java

```diff
@@ -0,0 +1,113 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.trace.propagation;
+
+import io.opentelemetry.context.propagation.HttpTextFormat.Setter;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.TraceFlags;
+import io.opentelemetry.trace.TraceId;
+import io.opentelemetry.trace.TraceState;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+
+@State(Scope.Thread)
+public class HttpTraceContextInjectBenchmark {
+
+  @Param({
+    "905734c59b913b4a905734c59b913b4a",
+    "21196a77f299580e21196a77f299580e",
+    "2e7d0ad2390617702e7d0ad239061770",
+    "905734c59b913b4a905734c59b913b4a",
+    "68ec932c33b3f2ee68ec932c33b3f2ee"
+  })
+  public static String traceIdBase16;
+
+  @Param({
+    "9909983295041501",
+    "993a97ee3691eb26",
+    "d49582a2de984b86",
+    "776ff807b787538a",
+    "68ec932c33b3f2ee"
+  })
+  public static String spanIdBase16;
+
+  private byte sampledTraceOptionsBytes = 1;
+  private TraceFlags sampledTraceOptions = TraceFlags.fromByte(sampledTraceOptionsBytes);
+  private TraceState traceStateDefault = TraceState.builder().build();
+
+  private HttpTraceContext httpTraceContext;
+  private Map<String, String> carrier;
+  private Setter<Map<String, String>> setter =
+      new Setter<Map<String, String>>() {
+        @Override
+        public void set(Map<String, String> carrier, String key, String value) {
+          carrier.put(key, value);
+        }
+      };
+
+  @State(Scope.Thread)
+  public static class HttpTraceContextInjectState {
+
+    public TraceId traceId;
+    public SpanId spanId;
+
+    @Setup
+    public void setup() {
+      this.traceId = TraceId.fromLowerBase16(traceIdBase16, 0);
+      this.spanId = SpanId.fromLowerBase16(spanIdBase16, 0);
+    }
+  }
+
+  @Setup
+  public void setup() {
+    this.httpTraceContext = new HttpTraceContext();
+    this.carrier = new LinkedHashMap<>();
```

Comment:
Curious about the usage of a Linked hashmap here. Why not just a plain HashMap?

Reply:
Since we adding\removing only one value i don't think it have any difference at all with usual HashMap, i used linked because saw it here HttpTraceContextTest

Reply:
I believe that the Linked version has slightly higher overhead than the plain HashMap, so for benchmarks, I think we should prefer a HashMap over a LinkedHashMap, unless we have a need to preserve order (which it sounds like we don't).

Reply:
agree